sw/device/bsp/w25q/w25q.c

```diff
@@ -0,0 +1,1247 @@
+/*
+                              *******************
+******************************* C SOURCE FILE *****************************
+**                            *******************
+**
+** project  : X-HEEP
+** filename : w25q.c
+** version  : 1
+** date     : 1/11/2023
+**
+***************************************************************************
+**
+** Copyright (c) EPFL contributors.
+** All rights reserved.
+**
+***************************************************************************
+*/
+
+/***************************************************************************/
+/***************************************************************************/
+/**
+* @file   w25q.c
+* @date   1/11/2023
+* @brief  Source file of the W25Q-family flash memory driver.
+*/
+
+/****************************************************************************/
+/**                                                                        **/
+/*                             MODULES USED                                 */
+/**                                                                        **/
+/****************************************************************************/
+#include "w25q128jw.h"
+
+/* To manage addresses. */
+#include "mmio.h"
+
+/* To manage interrupts. */
+#include "fast_intr_ctrl.h"
+#include "csr.h"
+#include "stdasm.h"
+
+/* To manage DMA. */
+#include "dma.h"
+
+/* To get TX and RX FIFO depth */
+#include "spi_host_regs.h"
+
+/* To get the target of the compilation (sim or pynq) */
+#include "x-heep.h"
+
+/****************************************************************************/
+/**                                                                        **/
+/*                        DEFINITIONS AND MACROS                            */
+/**                                                                        **/
+/****************************************************************************/
+
+/**
+ * The flash is expecting the address in Big endian format, so a swap is needed
+ * in order to provide the MSB first.
+*/
+#define REVERT_24b_ADDR(addr) ((((uint32_t)(addr) & 0xff0000) >> 16) | ((uint32_t)(addr) & 0xff00) | (((uint32_t)(addr) & 0xff) << 16))
```

Comment:
can't you use the function we already have in `device/lib/base/bitfield.c` called `bitfield_byteswap32`??

Reply:
Oh, I did't know of the existance of this. As I have to still do a shift every time, I'm keeping the define.